src/ZenstruckFoundryBundle.php

```diff
@@ -192,13 +202,46 @@ public function configure(DefinitionConfigurator $definition): void
 
     public function loadExtension(array $config, ContainerConfigurator $configurator, ContainerBuilder $container): void // @phpstan-ignore missingType.iterableValue
     {
-        $container->registerForAutoconfiguration(Factory::class)
-            ->addTag('foundry.factory')
-        ;
+        $container->registerForAutoconfiguration(Factory::class)->addTag('foundry.factory');
+
+        $container->registerForAutoconfiguration(Story::class)->addTag('foundry.story');
+
+        $hookValidator = static function(\ReflectionClass $reflector, string $hookEventClass, object $attribute): void {
+            if (
+                !$reflector->hasMethod('__invoke')
+                || 1 !== \count($parameters = $reflector->getMethod('__invoke')->getParameters())
+                || !($typeFirstParameter = $parameters[0]->getType())
+                || (string) $typeFirstParameter !== $hookEventClass
+            ) {
+                throw new InvalidArgumentException(\sprintf('In order to be declared "#[%s]", the class "%s" must have an "__invoke" method with one unique parameter of type "%s".', (new \ReflectionClass($attribute))->getShortName(), $reflector->getName(), $hookEventClass));
```

Comment:
I wondered if I should introduce an interface, but, that would mean 3 interface

not sure what's the  best :shrug: